systemtest/src/test/java/io/strimzi/systemtest/metrics/MetricsST.java

```diff
@@ -228,33 +236,30 @@ void testClusterOperatorMetrics() {
             assertCoMetricNotNull("strimzi_reconciliations_failed_total", resource, clusterOperatorMetricsData);
         }
 
-        Pattern connectResponse = Pattern.compile("strimzi_resources\\{kind=\"Kafka\",} ([\\d.][^\\n]+)");
-        ArrayList<Double> values = MetricsUtils.collectSpecificMetric(connectResponse, clusterOperatorMetricsData);
-        assertThat(values.stream().mapToDouble(i -> i).sum(), is((double) 2));
+        assertCoMetricResources(Kafka.RESOURCE_KIND, 2);
+        assertCoMetricResourceState(Kafka.RESOURCE_KIND, CLUSTER_NAME, 1);
+        assertCoMetricResourceState(Kafka.RESOURCE_KIND, SECOND_CLUSTER, 1);
 
-        connectResponse = Pattern.compile("strimzi_resources\\{kind=\"KafkaBridge\",} ([\\d.][^\\n]+)");
-        values = MetricsUtils.collectSpecificMetric(connectResponse, clusterOperatorMetricsData);
-        assertThat(values.stream().mapToDouble(i -> i).sum(), is((double) 1));
+        assertCoMetricResources(KafkaBridge.RESOURCE_KIND, 1);
+        assertCoMetricResourceState(KafkaBridge.RESOURCE_KIND, BRIDGE_CLUSTER, 1);
 
-        connectResponse = Pattern.compile("strimzi_resources\\{kind=\"KafkaConnect\",} ([\\d.][^\\n]+)");
-        values = MetricsUtils.collectSpecificMetric(connectResponse, clusterOperatorMetricsData);
-        assertThat(values.stream().mapToDouble(i -> i).sum(), is((double) 1));
+        assertCoMetricResources(KafkaConnect.RESOURCE_KIND, 1);
+        assertCoMetricResourceState(KafkaConnect.RESOURCE_KIND, CLUSTER_NAME, 1);
 
-        connectResponse = Pattern.compile("strimzi_resources\\{kind=\"KafkaConnectS2I\",} ([\\d.][^\\n]+)");
-        values = MetricsUtils.collectSpecificMetric(connectResponse, clusterOperatorMetricsData);
-        assertThat(values.stream().mapToDouble(i -> i).sum(), is((double) 0));
+        assertCoMetricResources(KafkaConnectS2I.RESOURCE_KIND, 0);
+        assertCoMetricResourceState(KafkaConnectS2I.RESOURCE_KIND, CLUSTER_NAME, 0);
```

Comment:
if there is no resource of a specific kind, the corresponding `state_resource_state` metric should not exist. Do you see it existing but with 0 value?

Reply:
In case it's not exists, the check return 0, but it return 0 in case the resource is nto ready. I can try to rewrite it a little bit and check that metric is not expsoed in case the resource is nto created.

Reply:
Yeah exactly, they are two different things: metric doesn't exist vs metric 0 (resource not ready). I would test them differently.